pkg/watches/helpers.go

```diff
@@ -171,3 +175,75 @@ func getCscOwnerKey(labels map[string]string) *client.ObjectKey {
 
 	return &client.ObjectKey{Namespace: ownerNamespace, Name: ownerName}
 }
+
+// WatchProxyEvents adds watches for proxy events.
+func WatchProxyEvents(c controller.Controller, client cl.Client, owner string) error {
+	// We only care if the event came from the cluster proxy.
+	pred := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			if e.Meta.GetName() == proxy.ClusterProxyName {
+				return true
+			}
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// If the cluster operator is ever changed we should kick off an event.
+			if e.MetaOld.GetName() == proxy.ClusterProxyName || e.MetaNew.GetName() == proxy.ClusterProxyName {
```

Comment:
This check implies that the name changed which is not possible for kube resources. Checking either should be sufficient.

Reply:
This is good to know, thank you Aravindh.